### PR TITLE
State: Move AT middleware to a dynamic import

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

Currently, the automated transfer state middleware is always loaded, just to check the status on every occasion where we change the selected site or load one or all of the sites. However, I believe this action is not critical at all, so the related code could be deferred to a separate chunk. This offers a good opportunity for optimizing the main entry point chunk, because the AT state is the only one guilty for pulling the plugin state into the main entry point, and it isn't one of the smallest trees 😉  

So, in a nutshell, this PR moves the AT state middleware to a dynamic import, and that removes the AT and related plugins state from the main entry point.

#### Testing instructions

* cc @lamosty - #18896 contains an explanation on how to test this flow, but is that still up to date?
* Also, go through some `/plugins` routes for Jetpack / atomic sites and verify they still work well.

### CODE REVIEW

File: client/state/lib/middleware.js [hunk 1]
```diff
@@ -99,7 +85,11 @@ const handler = ( dispatch, action, getState ) => {
 		case SITE_RECEIVE:
 		case SITES_RECEIVE:
 			// Wait a tick for the reducer to update the state tree
-			setTimeout( () => {
+			setTimeout( async () => {
+				const { fetchAutomatedTransferStatusForSelectedSite } = await import(
+					/* webpackChunkName: "automated-transfer-state-middleware" */
+					'calypso/state/lib/automated-transfer-middleware'
+				);
```
Comment (jsnajdr) [inline client/state/lib/middleware.js]: Here we only want to defer the check to the next event loop tick, rather than doing it synchronously together with the `dispatch( { type: SITE_RECEIVE } )` action that triggered it. That's already achieved by the `await import` and the `setTimeout` becomes redundant.

**Review by jsnajdr — COMMENTED** [review]
The middleware and the data layer handler could be replaced by a `store.subscribe` handler with an async loop:

If the selected site has the `has_pending_automated_transfer` option enabled (there's an observer that checks for this), start fetching the AT status for the site every 3 seconds, until it becomes "complete" or "error". After "complete", record analytics and re-fetch the site object.

File: client/state/lib/middleware.js (hunk 1 above)
Comment (tyxla) [reply client/state/lib/middleware.js]: Good call 👍  Cleaned up in https://github.com/Automattic/wp-calypso/pull/57182/commits/b5f355902cb7921a6bf839fd3086010ac276c2e6

**Comment by tyxla** [comment]
> The middleware and the data layer handler could be replaced by a `store.subscribe` handler with an async loop:
> 
> If the selected site has the `has_pending_automated_transfer` option enabled (there's an observer that checks for this), start fetching the AT status for the site every 3 seconds, until it becomes "complete" or "error". After "complete", record analytics and re-fetch the site object.

Agreed! However, my goal wasn't to refactor the middleware; I rather wanted to reduce some load off the main entry point. How strongly do you feel about this suggestion? IMHO it's not a requirement for this PR, but let me know what you think.

File: client/state/lib/middleware.js [hunk 2]
```diff
@@ -97,12 +83,14 @@ const handler = ( dispatch, action, getState ) => {
...
-			// Wait a tick for the reducer to update the state tree
-			setTimeout( () => {
-				fetchAutomatedTransferStatusForSelectedSite( dispatch, getState );
-			}, 0 );
+		case SITES_RECEIVE: {
+			const { fetchAutomatedTransferStatusForSelectedSite } = await import(
+				/* webpackChunkName: "automated-transfer-state-middleware" */
+				'calypso/state/lib/automated-transfer-middleware'
+			);
+			fetchAutomatedTransferStatusForSelectedSite( dispatch, getState );
```
Comment (jsnajdr) [inline client/state/lib/middleware.js:91]: Here I'd rather do `import(...).then(({ fetch }) => fetch(...))` rather than turning the whole middleware handler `async`.

**Review by jsnajdr — APPROVED** [review]
> How strongly do you feel about this suggestion?

I meant it as a suggestion for a follow-up, totally not a blocker for this PR.

File: client/state/lib/middleware.js (hunk 2 above)
Comment (tyxla) [reply client/state/lib/middleware.js:91]: I kinda like it as it is. I generally favor the idea for async middleware. None of the middleware functionality here is critical, so eventually, it could all end up async. Can you think of any reasons why that would be a problem? Or is this just a stylistic preference of yours? I'm curious to hear your thoughts.

Comment (jsnajdr) [reply client/state/lib/middleware.js:91]: I like how the Promise version is self-contained inside the `case` statement and how the `await` doesn't leak outside it. That's all 🙂

Comment (tyxla) [reply client/state/lib/middleware.js:91]: I see. Well, it doesn't really feel like a leak to me, it's just syntax 😉 Also, it will come in handy if we decide to asynchronize the rest of the middleware there.
